packages/gridtools/package.py

```diff
@@ -37,8 +37,9 @@ class Gridtools(CMakePackage):
     depends_on('ncurses')
     depends_on('cmake@3.14.5:%gcc')
     depends_on('boost@1.67.0:')
-    depends_on('mpi',  type=('build', 'run'))
-    depends_on('cuda', when='+cuda', type=('build', 'run'))
+    depends_on('mpi',  type=('build', 'run'), when='~cuda')
+    depends_on('mpicuda',  type=('build', 'run'), when='+cuda')
+    depends_on('cuda', type=('build', 'run'), when='+cuda')
```

Comment:
This is only until version 2.0 I believe? Since in 2.0 gridtools became header only, so no building happens (dependent packages will build what they need).

Reply:
Yes exactly, might be a little bit useless.

Reply:
I think it would additionally need something like `when='<only until version 2.0> ... cuda'`.
Not sure if we could just completely remove support for gridtools versions before 2.0. I don't know if anyone still needs this.
Xavier or Carlos would probably know.

Reply:
Yes we still need this the mch version of cosmo is still using the old gridtools